src/components/Table/Table.vue

```diff
@@ -39,11 +51,15 @@
 export interface Field {
     key: string;
     label?: string;
+    sortable?: boolean;
+    asc?: boolean;
+    desc?: boolean;
 }
 </script>
 
 <script lang="ts" setup>
```

Comment:
I think what would be nice is the following:
```
// Parent
<Table
    v-model:fields="fields"
/>

// Table.vue
<script>
const props = defineProps({
    fields: {
        ...
    },
});

const fields = computed({
    get: () => props.fields,
    set: (f) => emit('update:fields'),
});

fields.name.asc = true; // Will trigger the emit.
</script>
```

Reply:
I know it looks a little messy but at least it is easier to use, the emit only returns sorts and you can pass it in the exact matter to collection or something.
if we have `update:fields` emit we should get and update fields and return it again to component so it can update sort,... and later we have to filter up the sorts to pass it to collection

Reply:
True, but this package has nothing to do with `model-collection`. In our custom component we can map the sorts to an object usable by the collection.